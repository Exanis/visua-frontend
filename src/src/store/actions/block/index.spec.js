import * as Actions from './index';

describe('Block Actions', () => {
    it('getBlockList', () => {
        expect(Actions.getBlockList(1, 'test')).toMatchSnapshot();
    });
    it('onBlockList', () => {
        expect(Actions.onBlockList(['first', 'second'])).toMatchSnapshot();
    });
    it('createBlock', () => {
        expect(Actions.createBlock('test-blockname', 'test-firstname', 'test-lastname', 'test-email', 'test-password', false, 1, 'test')).toMatchSnapshot();
    });
    it('onCreateBlockSuccess', () => {
        expect(Actions.onCreateBlockSuccess()).toMatchSnapshot();
    });
    it('onCreateBlockError', () => {
        expect(Actions.onCreateBlockError(['error'])).toMatchSnapshot();
    });
    it('resetCreateBlockState', () => {
        expect(Actions.resetCreateBlockState()).toMatchSnapshot();
    });
    it('editBlock', () => {
        expect(Actions.editBlock('test-uuid', 'test-blockname', 'test-firstname', 'test-lastname', 'test-email', 'test-password', false, 1, 'test')).toMatchSnapshot();
    });
    it('onEditBlockSuccess', () => {
        expect(Actions.onEditBlockSuccess()).toMatchSnapshot();
    });
    it('onEditBlockError', () => {
        expect(Actions.onEditBlockError(['error'])).toMatchSnapshot();
    });
    it('resetEditBlockState', () => {
        expect(Actions.resetEditBlockState()).toMatchSnapshot();
    });
    it('deleteBlock', () => {
        expect(Actions.deleteBlock('test-uuid', 1, 'test')).toMatchSnapshot();
    });
    it('onDeleteBlockSuccess', () => {
        expect(Actions.onDeleteBlockSuccess()).toMatchSnapshot();
    });
    it('resetDeleteBlockState', () => {
        expect(Actions.resetDeleteBlockState()).toMatchSnapshot();
    });
});
