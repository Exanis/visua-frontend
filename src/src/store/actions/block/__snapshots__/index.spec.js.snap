// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Actions createBlock 1`] = `
Object {
  "data": "test-blockname",
  "page": "test-firstname",
  "search": "test-lastname",
  "type": "block.create",
}
`;

exports[`Block Actions deleteBlock 1`] = `
Object {
  "page": 1,
  "search": "test",
  "type": "block.delete",
  "uuid": "test-uuid",
}
`;

exports[`Block Actions editBlock 1`] = `
Object {
  "data": "test-blockname",
  "page": "test-firstname",
  "search": "test-lastname",
  "type": "block.edit",
  "uuid": "test-uuid",
}
`;

exports[`Block Actions getBlockList 1`] = `
Object {
  "page": 1,
  "search": "test",
  "type": "block.list",
}
`;

exports[`Block Actions onBlockList 1`] = `
Object {
  "block": Array [
    "first",
    "second",
  ],
  "type": "block.onList",
}
`;

exports[`Block Actions onCreateBlockError 1`] = `
Object {
  "errors": Array [
    "error",
  ],
  "type": "block.create.error",
}
`;

exports[`Block Actions onCreateBlockSuccess 1`] = `
Object {
  "type": "block.create.ok",
}
`;

exports[`Block Actions onDeleteBlockSuccess 1`] = `
Object {
  "type": "block.delete.ok",
}
`;

exports[`Block Actions onEditBlockError 1`] = `
Object {
  "errors": Array [
    "error",
  ],
  "type": "block.edit.error",
}
`;

exports[`Block Actions onEditBlockSuccess 1`] = `
Object {
  "type": "block.edit.ok",
}
`;

exports[`Block Actions resetCreateBlockState 1`] = `
Object {
  "type": "block.create.reset",
}
`;

exports[`Block Actions resetDeleteBlockState 1`] = `
Object {
  "type": "block.delete.reset",
}
`;

exports[`Block Actions resetEditBlockState 1`] = `
Object {
  "type": "block.edit.reset",
}
`;
